pinterest/teletraan
# host type UI updates to support dropdown search and multiple pages

The #capacity-eng team built this dashboard that shows over-provisioned instances. Based on feedback from the observability team, capeng would like implement the following fixes to simplify workflow of selecting/changing instance types in Teletraan.

Introduce new UI updates for Host Types that will improve user experience and simplify the workflow for changing instance types this includes:
- Modifying the Host Types page to allow users to click "Next" and "Previous" to view more than 200 Host Types
- Addition of Network and Retired Columns to Host Type Page + Network field to Host Type creation
- Modification of Host Type selector to be search able and contain Network information on New Capacity and Cluster Configuration pages
- Greying out of retired instance types in Host Type selector

Test Plan
1. Navigate to the Cluster Configuration page 
2. Select the Host Type dropdown
3. Confirm that you can search/filter by provider name and abstract name
4. 
<img width="1723" alt="image" src="https://user-images.githubusercontent.com/69278532/209241281-a9fcf21c-8c07-4c64-b52e-020e39aa9267.png">
5. Confirm that retired host types are greyed out and unselectable
6. 
<img width="1722" alt="image" src="https://user-images.githubusercontent.com/69278532/209241310-966b98cd-eba2-4f5b-be1e-bbc2f6fb3d11.png">
7. Confirm that the Host Types are filtered when the Arch field is changed
<img width="1284" alt="image" src="https://user-images.githubusercontent.com/69278532/209241343-ce15375e-763e-4f23-96b6-1f03067d6e3d.png">
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/69278532/209241376-c5f18304-126b-4301-89fa-7d72af04b214.png">
9. Click the help button next to Host Types and confirm that the Network and Retired columns are displayed
<img width="1276" alt="image" src="https://user-images.githubusercontent.com/69278532/209241435-252cd207-43fa-4b2d-935f-d34c1b19baad.png">
11. Repeat these steps on the Cluster creation page
12. Navigate to the clouds/hosttypes page
13. Confirm that there are columns for Network and Retired
14. 
<img width="1709" alt="image" src="https://user-images.githubusercontent.com/69278532/209241558-a75c5af6-1147-4a6d-9ff2-9df36aec364a.png">
15. Click Add Host Types
16. Confirm there is an field to input Network
17. 
<img width="819" alt="image" src="https://user-images.githubusercontent.com/69278532/209241587-0ef41428-5008-418f-a9be-e316817257e3.png">
